File: app/src/main/java/hm/orz/chaos114/android/slideviewer/util/IntentUtil.java
package hm.orz.chaos114.android.slideviewer.util;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

public final class IntentUtil {
    private IntentUtil() {
        // no-op
    }

    public static void browse(Context context, String url) {
        browse(context, Uri.parse(url));
    }

    public static void browse(Context context, Uri uri) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_VIEW);
        intent.setData(uri);
        context.startActivity(intent);
    }
}

File: app/src/main/java/hm/orz/chaos114/android/slideviewer/util/IntentUtil.kt
package hm.orz.chaos114.android.slideviewer.util

import android.content.Context
import android.content.Intent
import android.net.Uri

object IntentUtil {

    fun browse(context: Context, url: String) {
        browse(context, Uri.parse(url))
    }

    fun browse(context: Context, uri: Uri) {
        Intent().let {
            it.action = Intent.ACTION_VIEW
            it.data = uri
            context.startActivity(it)
        }
    }
}
